### DISCUSSION TITLE
Incorrect slicing

### DISCUSSION THREAD

**Discussion by krishnaap** [pydata/xarray#6018] [Q&A] [answered]
Slicing shows incorrect dimensions
```imr=ds.sel(longitude=slice(50,100),latitude=slice(0,18))```

It is supposed to slice the lat and lon values between 0-18 and 50-100 respectively, But I m getting the following result 
>imr
>Out[165]: 
><xarray.Dataset>
>Dimensions:        (latitude: 0, longitude: 201)

How to resolve this issue?

**Comment by keewis** [comment, marked as answer]
what is the ordering of your data? Unfortunately, if your data is descending instead of ascending you'll have to change your `slice` objects:
```python
ds.sel(longitude=slice(50, 100), latitude=slice(18, 0))
```

**Reply by krishnaap** [reply]
Wow. That really helps, I didn't know the ordering of the data really matters! Thanks a lot.
